### PR TITLE
Nabeel & Josefin

### PR DESCRIPTION
**Pull request by Josse79**
This is a chatbot made by Nabeel & Josefin in our first pair programming project.
It was quite demanding with the Javascript but we are proud of the outcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,17 @@
 # Project Name
 
-Replace this readme with your own information about your project.
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+Chatbot project - pair programming.
+To interact in chat using Js by adding variables, functions, eventlisternes, conditionals etc.
```
Comment (SteppbySteph) [inline README.md:4]: Try to specify what you did in the project by avoiding using words such as "etc" so the code reviewer know what you have done all in all.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+We shared our experiences.
+Googled and looked into SO for other team mates questions and answers.
+Got inspired by other chatbots.
+We got the foundation and then built up the Js file step by step in order for both of us to follow the code.
+Mostly we did the pair programming with live share and got an exceptional result.
+If we had more time, we would have liked to sharpen up our Js knowledge.
```
Comment (SteppbySteph) [inline README.md:13]: Good description of the problem!

File: code/script.js [hunk 3]
```diff
@@ -1,47 +1,150 @@
 // All the DOM selectors stored as short variables
 const chat = document.getElementById('chat')
+const form = document.getElementById('name-form')
+const inputText = document.getElementById('name-input')
+const inputWrapper = document.getElementById('input-wrapper')
+const messageAudio = new Audio ("./assets/button_click.mp3");
```
Comment (SteppbySteph) [inline code/script.js:6]: Applauding you for using sound effects!

File: code/script.js [hunk 4]
```diff
@@ -1,47 +1,150 @@
...
+    </select>`
+
+  document.getElementById("select").addEventListener('change', () => { 
+    showMessage(`Please take me in a ${select.value}`, 'user')
+    setTimeout(rate,1000)
+  })
+}
+
+//Fourth function decide rate
+const rate = () => {
```
Comment (SteppbySteph) [inline code/script.js:108]: Very good on adding a conditional to your chatbot!

File: code/script.js [hunk 5]
```diff
@@ -1,47 +1,150 @@
...
+const handleNameInput =  (event) => {
+  event.preventDefault()
+  const name = inputText.value
+  showMessage(`My name is ${name}`, 'user')
+  inputText.value = ''
+  setTimeout(() => destination(), 1000)
+}
+
+//Second function choose of destination
+const destination = () => {
```
Comment (SteppbySteph) [inline code/script.js:59]: Your structure of the code is easy to follow in general. A small thing we noticed is that the code block between the curly bracket could be organized so it follows the same structure throughout the entire document. For example in this case the code block starts on a new line after the first curly bracket (row 50 and 59).  On row 68, 76, 84 they start in the same line. Maybe it doesn't look like this in the actual VS code and is rather an automatic feature of this website. Again this is just a minor thing that could be fixed but can also be disregarded.

File: code/style.css [hunk 6]
```diff
@@ -119,22 +127,87 @@ main {
...
+  height: 60px;
+  background: #C8C6C6;
+  width: 100%;
+  padding: 0 20px;
+  cursor: pointer;
+}
+
+
+/* Media queries */
+@media (max-width:667px) {
```
Comment (SteppbySteph) [inline code/style.css:199]: Yes! Good thing using media queries.  :)

File: code/style.css [hunk 7]
```diff
@@ -119,22 +127,87 @@ main {
...
+  transition: all 0.5s;
+ }
+ 
+ 
+ 
+ button:hover:after {
+  transform: skewX(-45deg) scale(1, 1);
+  -webkit-transition: all 0.5s;
+  transition: all 0.5s;
+ }
```
Comment (mathildakarlsson) [inline code/style.css:181]: Love the hover effect on the buttons! It's very clean yet stylish and smooth!

File: code/style.css [hunk 8]
```diff
@@ -90,18 +96,20 @@ main {
...
   font-weight: 600;
   font-size: 16px;
   line-height: 26px;
   padding: 16px 24px;
-  color: #0026ff;
+  background-color: #7e8186;
+  color: #fff;
   max-width: 40%;
+  -webkit-box-shadow: 5px 5px 15px 5px #000000; 
+  box-shadow: 5px 5px 15px 5px #000000;
```
Comment (mathildakarlsson) [inline code/style.css:107]: Nice touch with the box-shadows!

File: code/script.js [hunk 9]
```diff
@@ -1,47 +1,150 @@
...
+      setTimeout(vehicle, 2000)
+  })
+
+  document
+    .getElementById('mmxBtn')
+    .addEventListener('click', () => { showMessage('Malmö', 'user')
+    setTimeout(() => showMessage('Malmö is a nice place, choose your vehicle', 'bot'), 1000)
+    inputWrapper.innerHTML = ''
+    setTimeout(vehicle, 2000)
+  })
```
Comment (mathildakarlsson) [inline code/script.js:80]: Smart thing to put the whole code inside the document! It makes it easy to follow the steps in the code

File: code/script.js [hunk 10]
```diff
@@ -1,47 +1,150 @@
...
+  document
+    .getElementById('stoBtn')
+    .addEventListener('click', () => { showMessage('Stockholm', 'user')
+    setTimeout(() => showMessage('Stockholm is a nice place, choose your vehicle', 'bot'), 1000)
+    inputWrapper.innerHTML = ''
+    setTimeout(vehicle, 2000)
+  })
+}
+
+//Third function choose of vehicle
```
Comment (mathildakarlsson) [inline code/script.js:91]: Great commenting!

File: code/script.js [hunk 11]
```diff
@@ -1,47 +1,150 @@
...
+//Fourth function decide rate
+const rate = () => {
+  let price
+  if (select.value === 'Limo') {
+    price = '20000SEK'
+  } else if (select.value === 'Minibus') {
+    price = '5000SEK'
+  } else {
+    price = '10sek'
+    alert('Unfortunately your ride will be in a Rickshaw')
```
Comment (mathildakarlsson) [inline code/script.js:116]: Love this little feature  😂

File: code/script.js [hunk 12]
```diff
@@ -1,47 +1,150 @@
...
+    <button id="No">No</button>`
+
+    document.getElementById("Yes").addEventListener("click", () => {
+      showMessage('Yes', 'user')
+      setTimeout(order, 1000)
+    })
+    document.getElementById("No").addEventListener("click", () => {
+      location.reload()
+      return false  
+    })
```
Comment (mathildakarlsson) [inline code/script.js:132]: Nice touch with a possibility to reload and choose another destination and vehicle!

**Review by mathildakarlsson — COMMENTED** [review]
Great job Josefin and Nabeel! Your chatbot is stylish and looks professional, and at the same time it as humor (like the alert-thing) and features like sound-effects. Your code seems very well structured and is easy to follow. Great commenting! Maybe even some more comments would just be nice, even though it's not necessary. It definitely looks like you've met all the requirements for this project! Well done!

File: code/style.css (hunk 6 above)
Comment (Josse79) [reply code/style.css:199]: Super thanks Stephannie  for your review, we looked into your comments together and appreciate all of them. Best regards Nabeel & Josefin

File: code/style.css (hunk 7 above)
Comment (Josse79) [reply code/style.css:181]: Thanks Mathilda for your kind words and all your comments. We have been looking into your comments together and we appreciate all of it. Best regards Nabeel & Josefin

**Pull request closed without merging**